shopBuddy/shopBuddy/Item+Persistent.m

```diff
@@ -0,0 +1,77 @@
+//
+//  Item+Persistent.m
+//  shopBuddy
+//
+//  Created by Rachna Gupta on 7/21/22.
+//
+
+#import "Item+Persistent.h"
+#import "Parse/Parse.h"
+#import <objc/runtime.h>
+#import "ShoppingList.h"
+
+@implementation Item (Persistent)
+
+
+
+- (NSString *)objectID {
+    return objc_getAssociatedObject(self, @selector(objectID));
+}
+
+- (void)setObjectID:(NSString *)new_objectID {
+    objc_setAssociatedObject(self, @selector(objectID), new_objectID, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+}
+
++ (void)deleteItemFromList {
+    //TODO: delete functionality
+}
+
+- (void) syncPrices {
+    //TODO: prices
+}
+
+- (void) fetchSpecificItem {
+    //TODO: prices
+}
+/*
+- (void) saveItem {
+    //NSDictionary *dict = @{ @"barcode_number" : dictionary[@"barcode_number"], @"name" : dictionary[@"title"], @"images" : dictionary[@"images"], @"brand" : dictionary[@"brand"], @"item_description" : dictionary[@"description"]};
+    //PFObject *new_object = [PFObject objectWithClassName:@"Item" dictionary:dict];
+    Item *const newItem = [Item hydrateItemFromPFObject:new_object];
+    completion(newItem);
+    
+}
+ */
+
++ (void) createObject:(PFObject *)object {
+    [object saveInBackground];
+}
```

Comment:
Move to callsite and you should use `[object saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error){...}]` instead as described in https://docs.parseplatform.org/ios/guide/#saving-objects.

Read comment in similar method in `Shopping+Presistent.m` for additional recommendations.